### DISCUSSION TITLE
[Bug] banyandb start with standalone.Application closed due to ETCD grpc timeout

### DISCUSSION THREAD

**Discussion by huogithub** [apache/skywalking#12544] [Q&A]
banyandb  0.6.1
oap: 10.0.1
    storage:banyandb
    configuration: none

logs:
{"level":"debug","module":"DATA","name":"pprof-service","registered":5,"total":6,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"DATA","name":"metric-service","registered":6,"total":6,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"LIAISON","name":"metadata","registered":1,"total":5,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"LIAISON","name":"grpc","registered":2,"total":5,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"LIAISON","name":"liaison-http","registered":3,"total":5,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"LIAISON","name":"pprof-service","registered":4,"total":5,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"debug","module":"LIAISON","name":"metric-service","registered":5,"total":5,"time":"2024-08-19T03:58:51Z","message":"register flags"}
{"level":"info","time":"2024-08-19T03:58:51Z","message":"starting as a standalone server"}
{"level":"info","module":"STANDALONE","time":"2024-08-19T03:58:51Z","message":"started"}
...
...
{"level":"info","module":"ETCD","time":"2024-08-19T04:00:08Z","message":"revoking lease 7587880826434867204"}
{"level":"error","module":"ETCD","error":"context deadline exceeded","time":"2024-08-19T04:00:17Z","message":"failed to revoke lease 7587880826434867204"}
{"level":"error","module":"LIAISON-GRPC","error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded","written":"<nil>","time":"2024-08-19T04:00:35Z","message":"failed to receive message"}
{"level":"error","module":"LIAISON-GRPC","error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded","written":"<nil>","time":"2024-08-19T04:00:36Z","message":"failed to receive message"}

**Comment by huogithub** [comment]
![企业微信截图_17240655094124](https://github.com/user-attachments/assets/c6544210-4f1c-45f0-b69d-f9f9ff0f62b2)

**Reply by hanahmily** [reply]
"137" may be related to OOM issues. Can you use `docker inspect` to gather more details about this failure?

The errors in the logs indicate that clients are closing the gRPC streams and do not indicate issues with the server.

**Reply by huogithub** [reply]
Are you saying there's not enough memory，but no problem using ES，Recently switched to Banyandb 0.6.1 as storage.
mounting volumes can solve the problem?
![image](https://github.com/user-attachments/assets/079c0bd9-73e3-4c3f-978f-0e33b1805d06)

**Reply by hanahmily** [reply]
The "banyandb" container lacks persistent storage mechanisms, risking data loss upon restart or recreation. You should mount a volume to banyandb to address this issue.

**Reply by huogithub** [reply]
How to set time zone Installation On Docker？
How  Configure MAX memory On Docker？

**Reply by huogithub** [reply]
![Uploading image.png…]()

**Reply by hanahmily** [reply]
> How to set time zone Installation On Docker？ How Configure MAX memory On Docker？

These questions are beyond the scope of banyandb. Please search for the answers yourself. By the way, your latest image failed to be uploaded.

**Reply by huogithub** [reply]
> > How to set time zone Installation On Docker？ How Configure MAX memory On Docker？
> 
> These questions are beyond the scope of banyandb. Please search for the answers yourself. By the way, your latest image failed to be uploaded.

banyandb  docker image  not Supported time zone setting ?

**Reply by huogithub** [reply]
![image](https://github.com/user-attachments/assets/c43a61eb-43b6-4a21-8553-ccae063824de)

**Reply by wu-sheng** [reply]
We don't recommend and not see the point of changing TZ of DB container. Although we don't limit it either.

**Comment by huogithub** [comment]
![image](https://github.com/user-attachments/assets/20b3d54c-414d-4b3e-bd9a-8f8978566125)


[
    {
        "Id": "8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf",
        "Created": "2024-08-14T03:02:39.747646692Z",
        "Path": "/banyand",
        "Args": [
            "standalone"
        ],
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 137,
            "Error": "",
            "StartedAt": "2024-08-19T10:21:22.901138988Z",
            "FinishedAt": "2024-08-20T01:21:55.639161196Z"
        },
        "Image": "sha256:003d77d52ca623e952c406c87342409472005affe56a7de2c3b084f867fc7e06",
        "ResolvConfPath": "/var/lib/docker/containers/8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf/hostname",
        "HostsPath": "/var/lib/docker/containers/8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf/hosts",
        "LogPath": "/var/lib/docker/containers/8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf/8e43daa976d59869d8666ac55167319165eff6087d96714f91e7679d84ba9faf-json.log",
        "Name": "/banyandb",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {
                    "max-file": "3",
                    "max-size": "500m"
                }
            },
            "NetworkMode": "skywalking_default",
            "PortBindings": {
                "17912/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "17912"
                    }
                ],
                "17913/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "17913"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": [],
            "ConsoleSize": [
                0,
                0
            ],
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/ab93cbf4464461070bd5e1fecce7503ad8672c5a31f6a270304401f492ee8932-init/diff:/var/lib/docker/overlay2/55a9fd7e83824a69c93a25ef3de62adeda0d9582894a5b3db53d7711ab5c1000/diff:/var/lib/docker/overlay2/343fe378d9bd914b6d941a7dcdcd1bba242570d21c72b5c9249b3c2e8ad32840/diff:/var/lib/docker/overlay2/18f06f6ed0f24432f64c3f2513b7c98a1334237aca59845aa816322e2ac43b43/diff:/var/lib/docker/overlay2/c1d03f37e3520190e07fb2d95245dbf3927081b1aded23aafedd01c5fab26aad/diff",
                "MergedDir": "/var/lib/docker/overlay2/ab93cbf4464461070bd5e1fecce7503ad8672c5a31f6a270304401f492ee8932/merged",
                "UpperDir": "/var/lib/docker/overlay2/ab93cbf4464461070bd5e1fecce7503ad8672c5a31f6a270304401f492ee8932/diff",
                "WorkDir": "/var/lib/docker/overlay2/ab93cbf4464461070bd5e1fecce7503ad8672c5a31f6a270304401f492ee8932/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "8e43daa976d5",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "17912/tcp": {},
                "17913/tcp": {},
                "6060/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "standalone"
            ],
            "Image": "apache/skywalking-banyandb:0.6.1",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/banyand"
            ],
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "51e92fb3299d685a966b07a07a7ad1780453e81c9c62b34ce569ae425df6c550",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "skywalking",
                "com.docker.compose.service": "banyandb",
                "com.docker.compose.version": "1.24.1"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "a7f34979513bb316769a7ba57eb67130580a1211f8c0e9ec58dcea088a35a699",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/a7f34979513b",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "skywalking_default": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "banyandb",
                        "8e43daa976d5"
                    ],
                    "NetworkID": "1399436e9a9725f36e84d688cd4ff4934386ed2a86018dd84181d8919ef4bec1",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
